onnxruntime/core/providers/cpu/tensor/grid_sample.cc

```diff
@@ -134,113 +160,203 @@ Status GridSample<T>::Compute(OpKernelContext* context) const {
   const auto& input_dims = input->Shape();
   const auto& grid_dims = grid->Shape();
 
-  if (input_dims.NumDimensions() != 4 || grid_dims.NumDimensions() != 4) {
-    return Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT, "Only 4-D tensor is supported");
-  }
+  int64_t data_dims = input_dims.NumDimensions() - 2;
+  ORT_ENFORCE(static_cast<int64_t>(grid_dims.NumDimensions()) == data_dims + 2,
+              "grid dimensions must be ", data_dims + 2, "for input dimension of ", data_dims);
+
+  ORT_ENFORCE(grid_dims[grid_dims.NumDimensions() - 1] == data_dims,
+              "Last dimension of grid: ", grid_dims[grid_dims.NumDimensions() - 1], ", expect ", data_dims);
+
+  ORT_ENFORCE(input_dims.NumDimensions() == 4 || input_dims.NumDimensions() == 5, "Only 4-D or 5-D tensor is supported");
 
   auto N = input_dims[0];
   auto C = input_dims[1];
-  auto H_in = input_dims[2];
-  auto W_in = input_dims[3];
-  auto H_out = grid_dims[1];
-  auto W_out = grid_dims[2];
   ORT_ENFORCE(grid_dims[0] == N, "Grid batch size ", grid_dims[0], " does not match input batch size ", N);
-  ORT_ENFORCE(grid_dims[3] == 2, "Last dimension of grid: ", grid_dims[3], ", expect 2");
 
-  TensorShape Y_shape = {N, C, H_out, W_out};
-  auto& Y = *context->Output(0, Y_shape);
-  // Return early if the output tensor is going to be of size 0
-  if (Y.Shape().Size() == 0) {
-    return Status::OK();
+  if (input_dims.NumDimensions() == 5) {
+    ORT_ENFORCE(mode_ != Cubic, "Only support GridSample Cubic mode in 4-D cases.");
```

Comment:
It is part of onnx standard? Any plan to support it?

Reply:
Good catch! 
It is part of standard for completeness. However, in practice, Cubic interpolation in 3d is unnecessarily complex and costly. Torch does not support this case either: https://pytorch.org/docs/stable/generated/torch.nn.functional.grid_sample.html (Note: mode='bicubic' supports only 4-D input). This case will not exist in any real world models.